Annotations/RoutingTable.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * This file is part of DoctrineRestDriver.
+ *
+ * DoctrineRestDriver is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * DoctrineRestDriver is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DoctrineRestDriver.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Circle\DoctrineRestDriver\Annotations;
+
+use Circle\DoctrineRestDriver\Validation\Assertions;
+use Doctrine\Common\Annotations\AnnotationReader;
+
+/**
+ * Contains all routing information about all entities
+ *
+ * @author    Tobias Hauck <tobias@circle.ai>
+ * @copyright 2015 TeeAge-Beatz UG
+ */
+class RoutingTable {
+
+    /**
+     * @var array
+     */
+    private $routingTable = [];
+
+    /**
+     * @var AnnotationReader
+     */
+    private $reader;
+
+    /**
+     * @var array
+     */
+    private $annotations = [
+        'post'   => 'Circle\DoctrineRestDriver\Annotations\Insert',
+        'put'    => 'Circle\DoctrineRestDriver\Annotations\Update',
+        'get'    => 'Circle\DoctrineRestDriver\Annotations\Select',
+        'delete' => 'Circle\DoctrineRestDriver\Annotations\Delete',
+        'getAll' => 'Circle\DoctrineRestDriver\Annotations\Fetch'
+    ];
+
+    /**
+     * RoutingTable constructor
+     *
+     * @param array $entities
+     *
+     * @SuppressWarnings("PHPMD.StaticAccess")
+     */
+    public function __construct(array $entities) {
+        Assertions::assertHashMap('entities', $entities);
+
+        $this->reader = new AnnotationReader();
+
+        $aliases            = array_flip($entities);
+        $this->routingTable = array_reduce($entities, function ($carry, $namespace) use ($aliases) {
```

Comment:
das mit dem reduce in dem reduce kommt mir noch falsch vor, aber ich versteh auch grad nit genau was du machs, also du reduced entities (en array) auf en routingtable, was en auch en array der selben größe is (jede entity hat ein routing) => das is en map und das andere holt dann über die annotations die routes - aber warum du da das geflippte array brauchs, verstehe ich nich (kann aber auch mit php zu tun haben :D). Wenn dieses gedöns da is, um die keys richtig zu setzen (was ich vermute), dann mach das lieber schrittweise:

* ers die values als liste nehmen und zu den values transformieren die du brauchs
* dann die liste nehmen und auf die datenstruktur reducen, die du brauchst

aber evtl is es auch was anderes, ich bräuchte da input